### PR TITLE
ARM64-SVE: Always convert to mask for mask APIs

### LINKED ISSUE #104836: JIT: SVE Cleanup - In the importer, always `ConvertToMask` on SVE APIs that require a mask.
Discussion: https://github.com/dotnet/runtime/pull/104704#discussion_r1674822469

Code:
```cpp
        if (!varTypeIsMask(op))
        {
            op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);
        }
```

The idea is that we shouldn't have to check for `varTypeIsMask(op)`; meaning at this point, `op` should be returning `TYP_SIMD`. Therefore, we should always add a `ConvertToMask` node:
```cpp
        assert(varTypeIsSIMD(op));
        op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);
```

### LINKED ISSUE #104841: JIT: SVE Cleanup - Simplify `emitIns_Mov` code when emitting RMW intrinsics
In `hwintrinsiccodegenarm64.cpp`, we see the following code which handles RMW intrinsics:
```cpp
                if (targetReg != op2Reg)
                {
                    assert(targetReg != op1Reg);
                    GetEmitter()->emitIns_Mov(INS_sve_mov, emitTypeSize(node), targetReg, op2Reg, /* canSkip */ true);
                }
```
It can be simplified to just:
```cpp
                assert(targetReg != op1Reg);
                GetEmitter()->emitIns_Mov(INS_sve_mov, emitTypeSize(node), targetReg, op2Reg, /* canSkip */ true);
```
Because `emitIns_Mov` will not emit a `mov` if `targetReg == op2Reg`.

There are similar ones as well that are not `INS_sve_mov` related, such as:
```cpp
                if (targetReg != op2Reg)
                {
                    assert(targetReg != op1Reg);
                    assert(targetReg != op3Reg);
                    GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op2Reg,
                                              /* canSkip */ true);
                }
```
which can be:
```cpp
                assert(targetReg != op1Reg);
                assert(targetReg != op3Reg);
                GetEmitter()->emitIns_Mov(INS_mov, emitTypeSize(node), targetReg, op2Reg,
                                          /* canSkip */ true);
```
---
As a consideration, not part of this issue, but perhaps adding a new function `emitIns_MovIfNeeded` that simply wraps `emitIns_Mov` with the `canSkip` set to `true` would be clearer.

### PR DESCRIPTION
**Pull request by a74nh**
Fixes #104836

SVE Stress tests all pass

### CODE REVIEW

**Comment by a74nh** [comment]
@dotnet/arm64-contrib @kunalspathak @amanasifkhalid

File: src/coreclr/jit/hwintrinsic.cpp [hunk 1]
```diff
@@ -2261,11 +2261,9 @@ GenTree* Compiler::impHWIntrinsic(NamedIntrinsic        intrinsic,
...
 #if defined(FEATURE_MASKED_HW_INTRINSICS) && defined(TARGET_ARM64)
-    auto convertToMaskIfNeeded = [&](GenTree*& op) {
-        if (!varTypeIsMask(op))
-        {
-            op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);
-        }
+    auto convertToMask = [&](GenTree*& op) {
+        assert(varTypeIsSIMD(op));
+        op = gtNewSimdCvtVectorToMaskNode(TYP_MASK, op, simdBaseJitType, simdSize);
     };
```
Comment (tannergooding) [inline src/coreclr/jit/hwintrinsic.cpp:2267]: We can get rid of `convertToMask` entirely and just use `gtNewSimdCvtVectorToMaskNode` directly. That is more clear, removed unnecessary overhead, and doesn't lose any functionality (an `assert(varTypeIsSIMD(op)` already exists in `gtNewSimdCvtVectorToMaskNode`).

Comment (a74nh) [reply src/coreclr/jit/hwintrinsic.cpp:2267]: I considered getting rid of `convertToMask()` but I didn't realise the assert was inside `gtNewSimdCvtVectorToMaskNode()`. Will fix.

Comment (a74nh) [reply src/coreclr/jit/hwintrinsic.cpp]: `convertToMask()` removed.

**Review by amanasifkhalid — APPROVED** [review]
LGTM, thanks!

**Pull request merged**